### PR TITLE
Add JSON support

### PR DESCRIPTION
**Pull request by zheng-da**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: python/graphstorm/gconstruct/construct_graph.py [hunk 1]
```diff
@@ -39,8 +39,58 @@
...
+    with open(data_file, 'r', encoding="utf8") as json_file:
+        data_records = []
+        for line in json_file.readlines():
+            record = json.loads(line)
+            data_records.append(record)
+
+    data = {key: [] for key in data_fields}
+    for record in data_records:
+        for key in data_fields:
+            data[key].append(record[key])
```
Comment (classicsong) [inline python/graphstorm/gconstruct/construct_graph.py:68]: How do we distinguish labels and features?

File: tests/end2end-tests/data_process/data_gen.py [hunk 2]
```diff
@@ -56,6 +44,8 @@ def write_data_parquet(data, data_file):
 node_id3_str = np.array([str(nid) for nid in node_id3])
 node_data3 = {
```
Comment (classicsong) [inline tests/end2end-tests/data_process/data_gen.py:45]: Can you add some docstring about how does the test graph looks like?

File: tests/unit-tests/test_construct_graph.py [hunk 3]
```diff
@@ -41,6 +41,22 @@ def test_parquet():
     assert "data2" not in data1
     assert np.all(data1['data1'] == data['data1'])
 
+def test_json():
+    from graphstorm.gconstruct import write_data_json, read_data_json
+    handle, tmpfile = tempfile.mkstemp()
```
Comment (classicsong) [inline tests/unit-tests/test_construct_graph.py:46]: Will the tmpfile be deleted after the test?

File: python/graphstorm/gconstruct/construct_graph.py (hunk 1 above)
Comment (zheng-da) [reply python/graphstorm/gconstruct/construct_graph.py:68]: for read/write functions, we don't distinguish them.

File: python/graphstorm/gconstruct/construct_graph.py [hunk 4]
```diff
@@ -39,8 +39,58 @@
...
+    for key in data:
+        data[key] = np.array(data[key])
+    return data
+
+def write_data_json(data, data_file):
```
[the remark this answers is not in the document]
Comment (zheng-da) [reply python/graphstorm/gconstruct/construct_graph.py:73]: i think this is for consistency. i plan to move all read/write functions to another file in another PR.

File: tests/end2end-tests/data_process/data_gen.py [hunk 5]
```diff
@@ -96,7 +86,7 @@ def split_data(data, num):
 for i, node_data in enumerate(split_data(node_data2, 10)):
     write_data_parquet(node_data, os.path.join(in_dir, f'node_data2_{i}.parquet'))
 for i, node_data in enumerate(split_data(node_data3, 10)):
-    write_data_parquet(node_data, os.path.join(in_dir, f'node_data3_{i}.parquet'))
+    write_data_json(node_data, os.path.join(in_dir, f'node_data3_{i}.json'))
```
[the remark this answers is not in the document]
Comment (zheng-da) [reply tests/end2end-tests/data_process/data_gen.py:89]: i want to make test all different cases. I just pick node3 to store the data with json. it's not necessary to store all node data with the same format.

**Review by classicsong — APPROVED** [review]

File: tests/end2end-tests/data_process/data_gen.py (hunk 2 above)
Comment (zheng-da) [reply tests/end2end-tests/data_process/data_gen.py:45]: i don't think there is any particular purpose of each node type. I create multiple node types to handle different cases, for example, to store node data with different formats, with or without node features, with or without labels, features of different types, node features with different dimensions. you can easily see this kind of information in the test code when node data are defined. I'm not sure what information should be documented.

**Pull request merged**